src/sentry/static/sentry/app/components/dropdownAutoComplete.jsx

```diff
@@ -0,0 +1,163 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+import AutoComplete from './autoComplete';
+import Input from '../views/settings/components/forms/styled/input';
+
+class DropdownAutoComplete extends React.Component {
+  static propTypes = {
+    items: PropTypes.array,
+    isOpen: PropTypes.bool,
+    onSelect: PropTypes.func,
+    children: PropTypes.func,
+  };
+
+  static defaultProps = {
+    isOpen: false,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isOpen: this.props.isOpen,
+    };
+  }
+
+  toggleMenu = () => this.setState({isOpen: !this.state.isOpen});
+
+  openMenu = e => this.setState({isOpen: true});
+
+  onSelect = selectedItem => {
+    this.setState({selectedItem});
+    if (this.props.onSelect) this.props.onSelect(selectedItem);
+    this.toggleMenu();
+  };
+
+  ungroupItems = itemsToUngroup =>
+    itemsToUngroup.reduce((accumulator, {group, items}) => {
+      return [
+        ...accumulator,
+        {...group, group: true},
+        ...items.map(g => ({...g, groupedWith: group.value})),
+      ];
+    }, []);
+
+  filterItems = (items, inputValue) =>
+    items.filter(
+      i => i.group || i.value.toLowerCase().indexOf(inputValue.toLowerCase()) > -1
+    );
+
+  removeUnusedGroups = items =>
```

Comment:
if its all the same, this will be simpler if it runs on the grouped item objects